organization/search_indexes.py

```diff
@@ -0,0 +1,205 @@
+
+        self.prepared_data['content'] = self._prepare_content(self.prepared_data)
+        self.prepared_data['countries'] = self._prepare_countries(self.prepared_data)
+
+        return self.prepared_data
+
+    def _prepare_countries(self, prepared_data):
+        countries = set()
+
+        for division in prepared_data['division_ids']:
+            countries.update([country_name(division)])
+
+        return list(countries)
+
+    def prepare_published(self, object):
+        parents = object.parent_organization.all()
+
+        return all([p.value.published for p in parents])
+
+    def prepare_division_ids(self, object):
+        division_ids = set()
+
+        # Start by getting the division ID recorded for this org
+        org_division_id = object.division_id.get_value()
+        if org_division_id:
+            division_ids.update([org_division_id.value])
+
+        # Grab foreign key sets
+        emplacements = object.emplacementorganization_set.all()
+
+        for emp in emplacements:
+            emp = emp.object_ref
+            site = emp.site.get_value()
+
+            if site:
+                exactloc_name = site.value.name
+                emp_division_id = site.value.division_id
+
+                if emp_division_id:
+                    division_ids.update([emp_division_id])
+
+        return list(division_ids)
+
+    def prepare_start_date(self, object):
+        return self._format_date(object.firstciteddate.get_value())
+
+    def prepare_end_date(self, object):
+        return self._format_date(object.lastciteddate.get_value())
+
+    def prepare_open_ended(self, object):
+        open_ended = object.open_ended.get_value()
+
+        if open_ended:
+            return open_ended.value
+
+    def prepare_name(self, object):
+        return object.name.get_value().value
+
+    def prepare_parent_names(self, object):
+        parent_names = []
+
+        for parent in object.parent_organization.all():
+            # `parent_organization` returns a list of CompositionChilds,
+            # so we have to jump through some hoops to get the foreign
+            # key value
+            composition = parent.object_ref
+            parent = composition.parent.get_value().value
+            parent_name = parent.name.get_value().value
+            parent_names.append(parent_name)
+
+        return parent_names
+
+    def prepare_memberships(self, object):
+        memberships = []
+
+        for membership in object.membershiporganizationmember_set.all():
+            # Similar to parents, we have to traverse the directed graph
+            # in order to get the entities we want
+            org = membership.object_ref.organization.get_value().value
+
+            if org.name.get_value():
+                memberships.append(org.name.get_value())
+
+        return memberships
+
+    def prepare_classifications(self, object):
+        return [cls.get_value().value for cls in object.classification.get_list()]
```

Comment:
i wonder if we can get a helper method to for all this resolving 

maybe something like

```python
def resolve(complex_object):
    return complex_object.get_value().value
```

Reply:
elaborate the function a bit

Reply:
If I felt more comfortable with the relationships, I'd say yes, but my grasp of the model is not firm enough to confidently abstract this out. I'd prefer to leave it as is for now.